src/core/mount.c

```diff
@@ -788,20 +788,28 @@ static int mount_spawn(Mount *m, ExecCommand *c, pid_t *_pid) {
         if (r < 0)
                 return r;
 
+        /* We must process /proc/self/mountinfo updates from
+         * the mount immediately, so record that it is pending.
+         */
+        UNIT(m)->manager->mount_pending_count++;
```

Comment:
I don't see what reading information about /tmp/hoge would be delayed - mount_pending_count should still be elevated at this point, even if there is a concurrent mount happening.

I put the increment there because I'm normally a kernel programmer and I expect things to happen concurrently.  Systemd is, of course, single-threaded so it is perfectly safe to leave the increment to the end of the function like you suggest, and not have to decrement on error.  I'll fix that.
Thanks.

Reply:
> I don't see what reading information about /tmp/hoge would be delayed - mount_pending_count should still be elevated at this point, even if there is a concurrent mount happening.

Indeed, you are right. Thank you.